TestProject/ClientApp/src/components/Post.js

```diff
@@ -21,8 +24,9 @@ function Post({ post, userLoggedIn, userProfile }) {
                 <p className="card-text">{post.content}</p>
                 {userLoggedIn && userProfile.id === post.author.id
                     && <div>
-                    <a href={`/posts/edit/${post.id}`} className="btn btn-primary btn-sm m-1">Edit</a>
-                    <button onClick={handleDelete} className="btn btn-danger btn-sm m-1">Delete</button>
+                    <Link to={`/posts/edit/${post.id}`} className="btn btn-primary btn-sm m-1">Edit</Link>
+                    <button onClick={() => { if (window.confirm("Are you sure you want to delete this post?")) handleDelete() }}
```

Comment:
Don't use `window.alert`/`window.confirm`, add a custom modal.
Don't use `if` in expressions like this, it's hardly readable. You can simply replace it with `&&`.